### PR TITLE
Add DGM `md5`

### PR DESCRIPTION
**Pull request by danielsun1106**

### CODE REVIEW

File: src/test/org/codehaus/groovy/runtime/DefaultGroovyMethodsTest.groovy [hunk 1]
```diff
@@ -310,4 +312,18 @@ class DefaultGroovyMethodsTest extends GroovyTestCase {
         assertTrue(DefaultGroovyMethods.implies(false, null))
         assertFalse(DefaultGroovyMethods.implies(true, null))
     }
+
+    void testMd5() {
+        try {
+            Assert.assertEquals("e99a18c428cb38d5f260853678922e03", StringGroovyMethods.md5("abc123"))
```
Comment (paulk-asert) [inline src/test/org/codehaus/groovy/runtime/DefaultGroovyMethodsTest.groovy]: StringGroovyMethods should be DefaultGroovyMethods or the DGM method should move? In any case, perhaps EncodingGroovyMethods might be better?

**Comment by paulk-asert** [comment]
Needs a Jira ticket

File: src/test/org/codehaus/groovy/runtime/DefaultGroovyMethodsTest.groovy (hunk 1 above)
Comment (danielsun1106) [reply src/test/org/codehaus/groovy/runtime/DefaultGroovyMethodsTest.groovy]: My bad... we should use `DefaultGroovyMethods`

**Comment by danielsun1106** [comment]
OK. I'll create a Jira ticket later

**Pull request closed without merging**